### ISSUE TITLE
isVisibleOnScreen's overflowHidden can false-negative with intermediate overflow: auto containers

### ISSUE THREAD

**Issue by dbjorge** [dequelabs/axe-core#3883]
### Product

axe-core

### Product Version

4.6.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

`isVisibleOnScreen`'s `overflowHidden` check currently treats elements as hidden if they are contained within an `overflow: hidden` ancestor whose bounding box does not overlap with the element under test.

However, this isn't quite accurate if an outer container uses `overflow: hidden` but contains an intermediate container which uses `overflow: auto`, which then contains the element under test. In this scenario, an element which is scrolled far out of view might be treated as "overflow-hidden" based on the outer container, even if it is scrollable into view via the intermediate container.

In the motivating example, this pattern of containers is used to implement "sticky header" functionality: `body` has `overflow: hidden` applied, the sticky header appears statically within the body, and then the rest of the page is rendered within a `overflow: auto` intermediate container. This results in false negatives against content "below the fold" in the intermediate container for any check whose `matches` includes a `isVisibleOnScreen` conditional.

#### Expectation

In the pattern described above, `isVisibleOnScreen` should treat an element scrolled out of view in the intermediate container the same as it would treat an element scrolled out of view in a page with no custom `overflow` styling.

#### Actual

In the pattern described above, `isVisibleOnScreen` treats elements scrolled out of view as "not visible", where it would have treated the same element scrolled out of view on a page not using such a pattern as "visible".

#### How to Reproduce
[Codepen demonstrating same content with different results depending on use of sticky header pattern](https://codepen.io/dbjorge/pen/VwBQOBO) - see repro steps embedded in page.

#### Additional context
The repro steps in the linked codepen use `label-content-name-mismatch` as a test case - it is one example of a rule which only matches elements that are `isVisibleOnScreen`, but probably not the only one.

**Comment by straker** [comment]
Thanks for the issue. Confirmed that this is the case.

**Comment by WilcoFiers** [comment]
To quote @michael-siek 

> ![Family Guy Mem, of Peter fighting window blinds, text saying "CSS"](https://media4.giphy.com/media/13FrpeVH09Zrb2/giphy.gif?cid=6104955et3rq7f623zx7cmsd3a9wgtl3nbx8ru0tw9uupdss&rid=giphy.gif&ct=g)

Well spotted Dan... endless "fun" we have with CSS here on axe-core.

**Comment by chutchins25** [comment]
Picking this up under #5226. Confirmed it still reproduces on 4.13.0 with the sticky-header pattern (`body { overflow: hidden }` → intermediate `overflow: auto` container → content below the fold): the below-fold element is treated as not-visible, so `label-content-name-mismatch` skips it (false negative), while an identical control element outside the pattern is correctly flagged.

**Root cause:** `overflowHidden` (`lib/commons/dom/visibility-methods.js`) collects the outer `overflow: hidden` ancestor via `getOverflowHiddenAncestors` and treats the element as hidden because its rect doesn't overlap that ancestor's rect. It doesn't account for an intermediate scrollable (`overflow: auto`/`scroll`) container between the element and the `overflow: hidden` ancestor — the element can be scrolled into view there, so it isn't actually clipped away.

**Proposed approach:** mirror the existing `hasPositionedAncestorBetween` guard — before treating an `overflow: hidden` ancestor as hiding the element, bail if a scrollable container sits between the element and that ancestor:

```js
// inside overflowHidden's nodes.some(node => { ... })
if (hasScrollableAncestorBetween(vNode, node)) {
  return false; // reachable by scrolling an intermediate container
}
```

**Two things I'd like your read on before coding:**

1. **Blast radius.** `overflowHidden` → `isVisibleOnScreen` feeds color-contrast, target-size (via `create-grid`), modal detection, the `autocomplete`/`link-in-text-block`/`is-visible` matchers, and `visible-virtual`. A change here can ripple into all of those, so I'd validate against each suite. Any known-fragile edge cases to watch (`overflow-x` vs `overflow-y`, nested scroll containers, `auto` that isn't actually scrollable)?
2. **Does this need to block #5226?** This is a general `isVisibleOnScreen` limitation affecting many rules, not specific to `label-content-name-mismatch`. Should it gate taking the rule out of experimental, or be handled as its own commons fix on its own timeline?

@straker / @WilcoFiers — appreciate any steer.

**Comment by chutchins25** [comment]
Put up a candidate fix in #5312 (draft): `overflowHidden` now bails when a scrollable (`overflow: auto/scroll`) container sits between the element and the `overflow:hidden` ancestor, since the element can be scrolled back into view.

Verified the sticky-header repro flips (`isVisibleOnScreen` `false`→`true`) with a regression test, and the affected suites (color-contrast, target-size/grid, the matchers, is-on-screen) are green locally.

Two things for your call: the mixed-axis edge case (`overflow-x:auto; overflow-y:hidden`) noted in the PR, and whether this should ride with #5226 or land on its own — it's general `isVisibleOnScreen` behavior, not specific to this epic. @straker / @dbjorge
